src/node/internal/crypto_keys.ts

```diff
@@ -276,7 +279,24 @@ export function createSecretKey(
   encoding?: string
 ): SecretKeyObject {
   validateKeyData(key, 'key');
-  if (typeof key === 'string') key = Buffer.from(key as string, encoding);
+  if (typeof key === 'string') {
+    key = Buffer.from(key as string, encoding);
+  } else if (isAnyArrayBuffer(key)) {
+    // We want the key to be a copy of the original buffer, not a view.
+    key = Buffer.from(new Uint8Array(key));
+  } else if (isArrayBufferView(key)) {
+    // We want the key to be a copy of the original buffer, not a view.
+    key = Buffer.from(key as any);
```

Comment:
This will fail with eslint. I think we need to fix our types for Buffer.from. Maybe a good TODO comment here would be sufficient for now.

Reply:
I'll worry about lint fixups when the PR is mostly finished.